### PR TITLE
Improve responsive layout on releases page

### LINKED ISSUE #1700: Collapse Table on Releases Page on Small and Medium Screens
### Description:
On the 'https://ocaml.org/releases' page, the content is currently displayed in the same row-based style as it appears on larger screens. 
This layout causes content to become hidden, requiring users to scroll through each release to view the complete content. This results in an unfavorable user experience. 
For a more comprehensive understanding of the problem, please consult the attached video demonstration.

https://github.com/ocaml/ocaml.org/assets/68381641/65b8d745-a71c-4b7f-9ef3-c7b0673637b3

### PR DESCRIPTION
**Pull request by RWUBAKWANAYO**
**Description:**

In this pull request, I made significant enhancements to the 'https://ocaml.org/releases' page to improve the user experience on small and medium screens. Instead of using HTML tables, I've replaced them with responsive div containers. This change ensures that the content is displayed optimally on a variety of devices and screen sizes.

In addition to the HTML structure updates, I've also fine-tuned the Tailwind CSS classes for these containers to ensure a proper responsive design. These improvements resolve the issue of content being hidden, eliminating the need for excessive scrolling.

Fixes #1700

### CODE REVIEW

File: src/ocamlorg_frontend/pages/releases.eml [hunk 1]
```diff
@@ -13,69 +13,74 @@ Layout.render
 </div>
 <div class="bg-default dark:bg-dark-default">
     <div class="py-10 lg:py-28">
-        <div class="container-fluid">
+        <div class="container-fluid min-w-full">
```
Comment (sabine) [inline src/ocamlorg_frontend/pages/releases.eml]: I would go for `container mx-auto`  instead, since container fluid wasn't a great choice here in the original design. `container` is more predictable in how it resizes. (Avoiding the fluid container here means we just have a few breakpoints to check regarding whether everything looks right.)

File: src/ocamlorg_frontend/pages/releases.eml [hunk 2]
```diff
@@ -13,69 +13,74 @@ Layout.render
...
+                    </h5>
+                    <form action="<%s Url.releases %>" method="GET">
+                        <%s! Forms.search_input
                         ~name:"q"
                         ~label:"Search for a release"
                         ~button_attrs:{|type="submit"|}
                         ""
                         %>
-                </form>
+                    </form>
```
Comment (sabine) [inline src/ocamlorg_frontend/pages/releases.eml]:
```suggestion
                </form>
```

unnecessary whitespace change

File: src/ocamlorg_frontend/pages/releases.eml [hunk 3]
```diff
@@ -13,69 +13,74 @@ Layout.render
...
-                        <%s! Icons.magnifying_glass "h-10 w-10" %>
+                    <div class="flex flex-col items-center justify-center pt-24 space-y-4 text-lighter">
+                        <div>
+                            <%s! Icons.magnifying_glass "h-10 w-10" %>
+                        </div>
+                        <div class="text-xl">No Releases found matching "
+                            <%s search %>"</div>
                     </div>
-                    <div class="text-xl">No Releases found matching "<%s search %>"</div>
-                </div>
```
Comment (sabine) [inline src/ocamlorg_frontend/pages/releases.eml]: indentation change here makes it hard to see what is changed. Please avoid introducing unnecessary whitespace changes to make reviewing easier.

File: src/ocamlorg_frontend/pages/releases.eml [hunk 4]
```diff
@@ -13,69 +13,74 @@ Layout.render
...
                                 </div>
-                            </td>
-                            <td class="py-4 px-6 font-medium align-top">
-                                <%s release.date %>
-                            </td>
-                            <td class="py-4 px-6 align-top">
-                                <a href="<%s Url.release release.version %>" class="text-primary-600 font-medium block">
+                            </div>
+                            <% releases |> List.iter (fun (release : Data.Release.t) -> %>
+                                <div class="flex flex-col lg:flex-row md:flex-col">
```
Comment (sabine) [inline src/ocamlorg_frontend/pages/releases.eml]:
```suggestion
                                <div class="flex flex-col lg:flex-row">
```

Would collapse on `md` because everything nicely left-aligns and people don't have the primary content floating to the right.

File: src/ocamlorg_frontend/pages/releases.eml [hunk 5]
```diff
@@ -13,69 +13,74 @@ Layout.render
...
+                                    <div class="py-4 px-6 font-semibold lg:w-2/3 md:w-full w-full">
```
Comment (sabine) [inline src/ocamlorg_frontend/pages/releases.eml]:
```suggestion
                                    <div class="py-4 px-6 font-semibold lg:w-2/3 w-full">
```

`w-full` includes `md:w-full` already, so no need for this class

File: src/ocamlorg_frontend/pages/releases.eml [hunk 6]
```diff
@@ -13,69 +13,74 @@ Layout.render
...
+                                        <div class="font-semibold">
+                                            <%s release.version %>
+                                        </div>
+                                        <div class="prose prose-orange font-normal">
+                                            <%s! release.highlights_html %>
+                                        </div>
+                                    </div>
+                                    <div class="lg:w-1/3 flex md:w-full w-full lg:flex-row md:flex-row md:justify-between flex-col">
```
Comment (sabine) [inline src/ocamlorg_frontend/pages/releases.eml]:
```suggestion
                                    <div class="lg:w-1/3 flex w-full lg:flex-row lg:justify-between flex-col">
```

File: src/ocamlorg_frontend/pages/releases.eml [hunk 7]
```diff
@@ -13,69 +13,74 @@ Layout.render
...
+                                        <div class="py-4 px-6 font-medium align-top lg:w-1/2 md:w-auto w-full">
```
Comment (sabine) [inline src/ocamlorg_frontend/pages/releases.eml]:
```suggestion
                                        <div class="py-4 px-6 font-medium align-top lg:w-1/2 w-full">
```

File: src/ocamlorg_frontend/pages/releases.eml [hunk 8]
```diff
@@ -13,69 +13,74 @@ Layout.render
...
+                                            <%s release.date %>
+                                        </div>
+                                        <div class="py-4 px-6 align-top lg:w-1/2 md:w-auto w-full">
```
Comment (sabine) [inline src/ocamlorg_frontend/pages/releases.eml]:
```suggestion
                                        <div class="py-4 px-6 align-top lg:w-1/2 w-full">
```

File: src/ocamlorg_frontend/pages/releases.eml [hunk 9]
```diff
@@ -13,69 +13,74 @@ Layout.render
...
-                                <a href="<%s Url.Package.overview "ocaml" ~version:release.version %>" class="text-primary-600 font-medium block">
+                                            <a href="<%s Url.Package.overview " ocaml " ~version:release.version %>" class="text-primary-600 font-medium block">
                                     OCaml Package
                                 </a>
-                                <a href="<%s Url.manual_with_version release.version %>" class="text-primary-600 font-medium block">
+                                            <a href="<%s Url.manual_with_version release.version %>" class="text-primary-600 font-medium block">
                                     Language Manual
                                 </a>
-                                <a href="<%s Url.api_with_version release.version %>" class="text-primary-600 font-medium block">
+                                            <a href="<%s Url.api_with_version release.version %>" class="text-primary-600 font-medium block">
```
Comment (sabine) [inline src/ocamlorg_frontend/pages/releases.eml]: unnecessary whitespace changes

File: src/ocamlorg_frontend/pages/releases.eml [hunk 10]
```diff
@@ -13,69 +13,74 @@ Layout.render
...
                                     Standard Library API
                                 </a>
-                            </td>
-                        </tr>
+                                        </div>
+                                    </div>
+                                </div>
+                                <% ); %>
+                        </div>
```
Comment (sabine) [inline src/ocamlorg_frontend/pages/releases.eml]: check that indentation matches the corresponding opening tag. I haven't checked, but looks suspicious.

**Review by sabine — COMMENTED** [review]
Hey, great work @RWUBAKWANAYO!

Very close to merge, just some smaller adjustments wrt collapsing at `md` and  cleaning up whitespace.

**Comment by sabine** [comment]
@RWUBAKWANAYO what is the status on this? Are you interested in continuing to work on this PR? If yes, I'll wait. If no, I'll free the issue for someone else to work on it.

In case help is needed, reach out in the OCaml Discord's #ocaml.org channel.

**Comment by RWUBAKWANAYO** [comment]
> @RWUBAKWANAYO what is the status on this? Are you interested in continuing to work on this PR? If yes, I'll wait. If no, I'll free the issue for someone else to work on it.
> 
> In case help is needed, reach out in the OCaml Discord's #ocaml.org channel.

@sabine, thank you for reaching out, I'm still interested and I'll make sure to apply the changes by end of tomorrow.

**Comment by sabine** [comment]
@RWUBAKWANAYO great, thanks!

File: src/ocamlorg_frontend/pages/releases.eml (hunk 10 above)
Comment (RWUBAKWANAYO) [reply src/ocamlorg_frontend/pages/releases.eml]: @sabine, The close tag corresponds to the code block used to check if there are releases to display and shows different templates based on whether releases exist or not.

**Comment by sabine** [comment]
@RWUBAKWANAYO this is very hard to review because you are making unnecessary whitespace/formatting changes, so that the entire file has to be checked line by line.

Can you please reduce the diff by ensuring that unchanged lines are indeed unchanged?

**Comment by RWUBAKWANAYO** [comment]
@sabine, I apologize for the review difficulty caused by changing the indentation from two tabs to one tab which affected the entire file, impacting both styles and code format. If you suggest that I should restrict the changes to styles only, I'm willing to immediately undo the modifications and focus on refining the styles without altering the indentation. Your input is highly appreciated.

**Comment by sabine** [comment]
Thanks for the contribution @RWUBAKWANAYO! :+1: Will merge after the CI checks complete.

**Review by sabine — APPROVED** [review]
